save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/UsersDetailsController.kt

```diff
@@ -109,36 +109,38 @@ class UsersDetailsController(
             UserSaveStatus.INVALID_NAME.message
         }
         .blockingMap {
-            val user: User = userRepository.findByName(newUserInfo.oldName ?: newUserInfo.name).orNotFound()
+            userRepository.findByIdOrNull(authentication.userId())
+        }
+        .switchIfEmptyToNotFound {
+            "Not found user in database for ${authentication.userId()}"
+        }
+        .blockingMap { user ->
+            val oldName = user.name.takeUnless { it == newUserInfo.name }
```

Comment:
didn't get why it's needed. usually username is not lastName or firstName

Reply:
I mean, since you are using the `takeUnless` probably, the `user.name` is some list, isnt it? `(oldName1, oldName2, oldName3 oldName4. .. , newName)` ?

So may be it will be convinient to have `oldName (lastOldName)` = `user.name.takeUnless { it == newUserInfo.name }`

and `firstOldName` = `user.name.first()`?

Reply:
`takeUnless ` -- return `this` or `null` if condition matches